src/core/postAttachments.ts

```diff
@@ -0,0 +1,52 @@
+import {tmpdir} from 'os';
+import {sep} from 'path';
+import joplin from 'api';
+import {Attachment} from '../model/attachment.model';
+import {AttachmentProperties} from '../model/attachmentProperties.mode';
+const fs = joplin.require('fs-extra');
+
+export class Attachments {
+    tempFolder: string = `${tmpdir}${sep}joplin-email-plugin${sep}`;
+    attachments: Attachment[];
+
+    constructor(attachments: Attachment[]) {
+        this.attachments = attachments.filter((e: Attachment)=>e.filename !== '');
+    }
+
+
+    async postAttachments(): Promise<AttachmentProperties[]> {
+        const attachmentsProp: AttachmentProperties[] = [];
+        const tempFolder = this.tempFolder;
+
+        try {
+            // for each attachment will create an actual file of the attachment and posting to Joplin.
+            for (let i = 0; i < this.attachments.length; i++) {
+                const {contentId, filename} = this.attachments[i];
+                const path = `${tempFolder}${this.attachments[i].filename}`;
+
+                // to create a file
+                fs.writeFileSync(path, Buffer.from(this.attachments[i].content));
+
+                // To post a file to Joplin
+                const resource = await joplin.data.post(
+                    ['resources'],
+                    null,
+                    {title: filename}, // Resource metadata
+                    [
+                        {
+                            path: path, // Actual file
+                        },
+                    ],
+                );
+
+                attachmentsProp.push({contentId: contentId, id: resource.id});
```

Comment:
might be worth cleaning up the temp file afterwards

Reply:
I tried deleting folders in this place, but it will slow down when there are a lot of messages that you want to convert because it will create and delete and so on. What I did is every time the plugin starts, it deletes the old attachments and starts creating a new temporary folder.

Reply:
Could potentially be a security risk, say, if you run the plugin in Joplin portable from a USB drive, your emails will be left on the computer.

Reply:
can't the delete job be a self-running/isolated background job? 
The "old" temp folder could be renamed, a new one created, old, now renamed, deleted in the background

Reply:
> Could potentially be a security risk, say, if you run the plugin in Joplin portable from a USB drive, your emails will be left on the computer.

Okay, I refacted the function to this mechanism.

 - Create a temp folder
- - Post all emails including attachments to Joplin.
- Remove a temp folder

Reply:
> can't the delete job be a self-running/isolated background job?
> The "old" temp folder could be renamed, a new one created, old, now renamed, deleted in the background

It's a sync function, which means I can't go to the next line after finishing this function(delete a temp folder).

Reply:
You could in theory use a web worker (or whatever the equivalent of threads in the JS world is called) but this looks like an overkill.

Reply:
> You could in theory use a web worker (or whatever the equivalent of threads in the JS world is called) but this looks like an overkill.

This looks very good since we can add all the notes and attachments posting functions in the other JS thread.
I will read more about web worker. and I will definitely add it if I notice its presence will be important.